docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,89 @@
+# Charm Architecture
+
+At its core, [Discourse](https://www.discourse.org/) is a [Ruby on Rails](https://rubyonrails.org/) application that integrates with [PostgreSQL](https://www.postgresql.org/) and [Redis](https://redis.io/).
+
+The charm design leverages the [sidecar](https://kubernetes.io/blog/2015/06/the-distributed-system-toolkit-patterns/#example-1-sidecar-containers) pattern to allow multiple containers in each pod with [Pebble](https://juju.is/docs/sdk/pebble) running as the workload container’s entrypoint.
+
+Pebble is a lightweight, API-driven process supervisor that is responsible for configuring processes to run in a container and controlling those processes throughout the workload lifecycle.
+
+Pebble `services` are configured through [layers](https://github.com/canonical/pebble#layer-specification), and the following containers represent each one a layer forming the effective Pebble configuration, or `plan`:
+
+1. A [Discourse](https://www.discourse.org/) container, consisting of a [Unicorn server](https://devcenter.heroku.com/articles/rails-unicorn) running the Discourse application along with the installed plugins.
+
+As a result, if you run a `kubectl get pods` on a namespace named for the Juju model you've deployed the Discourse charm into, you'll see something like the following:
+
+```bash
+NAME                            READY   STATUS    RESTARTS   AGE
+discourse-k8s-0                 2/2     Running   0          65m
+```
+
+This shows there are 2 containers - the one named above, as well as a container for the charm code itself.
+
+And if you run `kubectl describe pod discourse-k8s-0`, all the containers will have as Command ```/charm/bin/pebble```. That's because Pebble is responsible for the processes startup as explained above.
+
+## Containers
+
+Configuration files for the containers can be found in [the image directory of the charm repository](https://github.com/canonical/discourse-k8s-operator/tree/main/image).
+
+### Discourse
+
+Discourse is a Ruby on Rails application deployed on top of the [Unicorn server](https://devcenter.heroku.com/articles/rails-unicorn).
+
+The server is started in HTTP mode (port `3000`) serving all the content. Alongside it there's a standalone process running the [Prometheus Exporter Plugin for Discourse](https://github.com/discourse/discourse-prometheus) (port `9394`)
+
+The workload that this container is running is defined in the [Discourse dockerfile in the charm repository](https://github.com/canonical/discourse-k8s-operator/blob/main/discourse.Dockerfile).
+
+## Docker Images
+
+The image defined in [Discourse dockerfile](https://github.com/canonical/discourse-k8s-operator/blob/main/discourse.Dockerfile) in the charm repository are published to [Charmhub](https://charmhub.io/), the official repository of charms.
```

Comment:
I'd say "The image defined in the Discourse dockerfile" fwiw. If it's just one image I'd also say "in the charm repository is published".

Reply:
Done, thx